python/lsst/daf/butler/core/config.py

```diff
@@ -251,6 +251,7 @@ def __initFromFile(self, path):
                 self.__initFromYamlFile(uri.ospath)
         else:
             raise RuntimeError("Unhandled config file type:%s" % uri)
+        self.configFile = str(path)
```

Comment:
This might fix your butlerRoot woes.

Reply:
You are right that butler root was not correctly inferred or propagated from butler config file.  That is now fixed. 

I don't think there is a reasonable solution for default values, such as `curdir` for PosixDatastore, without making Butler capable of creating, at the very least, new buckets from scratch. I've always avoided doing it because the way permissions, regions, availability zones, VPCs etc. are set up matters. That means I could, eventually, be in a situation where someone might want to be able to specify these various details during Butler.makeRepo. I keep asking whether this is eventually wanted or not, but insofar it feels like a "no" to me.

Reply:
I don't really follow you. ButlerRoot exists as a convenience to allow you to say that the datastore is located relative to where the butler.yaml file was found. That's it. It's not meant to be about creating new buckets.  If the butler.yaml is on a local disk and the datastore is in S3 then your butler.yaml would not use butlerRoot for the datastore.root entry.

Reply:
Maybe I just got confused, up to recently `repoRelocation` would also assume current working directory was the default for `butlerRoot` and that made a lot of commands like `makeButlrRepo.py` work. I see this was changed end of November. I guess I was not paying attention.